controllers/nutanixmachine_controller.go

```diff
@@ -627,3 +634,22 @@ func (r *NutanixMachineReconciler) getMachineCategoryIdentifiers(rctx *nctx.Mach
 	}
 	return categoryIdentifiers
 }
+
+func (r *NutanixMachineReconciler) addBootTypeToVM(rctx *nctx.MachineContext, vmSpec *nutanixClientV3.VM) error {
+	bootType := rctx.NutanixMachine.Spec.BootType
+	if bootType != "" {
+		if bootType != string(infrav1.NutanixIdentifierBootTypeLegacy) && bootType != string(infrav1.NutanixIdentifierBootTypeUEFI) {
+			errorMsg := fmt.Errorf("%s boot type must be %s or %s but was %s", rctx.LogPrefix, string(infrav1.NutanixIdentifierBootTypeLegacy), string(infrav1.NutanixIdentifierBootTypeUEFI), bootType)
+			klog.Error(errorMsg)
+			return errorMsg
+		}
+
+		if bootType == string(infrav1.NutanixIdentifierBootTypeUEFI) {
```

Comment:
assuming default is legacy. can we document this in comment

Reply:
@yannickstruyf3 I let you add a comment in the code and also in the cluster-template.yaml and next we will merge the PR

Reply:
Done